src/arrayMethodJoin.js

```diff
@@ -5,8 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+    const last = this.length - 1;
+
+    let newSeparator = separator;
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    }
+
+    if (separator === null) {
+      newSeparator = 'null';
+    }
```

Comment:
```suggestion
```

This part is unnecessary, cause you can set default value for separator directly in function parameters.